catalog/spatial/registry/registry-policy-plugin/src/main/java/org/codice/ddf/registry/policy/RegistryPolicyPlugin.java

```diff
@@ -146,10 +148,30 @@ private void parsePermissionsFromString(List<String> permStrings,
 
     private PolicyResponse getWritePolicy(Metacard input, Map<String, Serializable> properties,
             Map<String, Set<String>> policy) {
-        HashMap<String, Set<String>> operationPolicy = new HashMap<>();
+
+        Map<String, Set<String>> operationPolicy = new HashMap<>();
+        Map<String, Set<String>> itemPolicy = new HashMap<>();
+        Map<String, Set<String>> securityAttributes = new HashMap<>();
+
         if (Requests.isLocal(properties) && input != null && (
                 RegistryUtility.isInternalRegistryMetacard(input)
                         || RegistryUtility.isRegistryMetacard(input))) {
+            Attribute securityLevel = input.getAttribute(RegistryObjectMetacardType.SECURITY_LEVEL);
+            if (securityLevel != null) {
+                securityLevel.getValues()
```

Comment:
I think you could use the Permissions.parsePermissionsFromString helper to do this for you.